protocols/secio/src/lib.rs

```diff
@@ -120,12 +119,53 @@ mod handshake;
 mod stream_cipher;
 mod structs_proto;
 
+pub use algo_support::{Digest, KeyAgreement};
+pub use stream_cipher::Cipher;
+
 /// Implementation of the `ConnectionUpgrade` trait of `libp2p_core`. Automatically applies
 /// secio on any connection.
 #[derive(Clone)]
 pub struct SecioConfig {
     /// Private and public keys of the local node.
-    pub key: SecioKeyPair,
+    pub(crate) key: SecioKeyPair,
+    pub(crate) agreements_prop: Option<String>,
+    pub(crate) ciphers_prop: Option<String>,
+    pub(crate) digests_prop: Option<String>
+}
+
+impl SecioConfig {
+    pub fn new(kp: SecioKeyPair) -> Self {
```

Comment:
Please add doc comments everywhere.

Also, it would eventually be nice to generate a key pair and create the corresponding config at once, but that's not for now.